File: JavaScript/7_reverseInteger.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    const INT_MAX = 2**31 - 1;
    const INT_MIN = -(2**31);
    let y = Math.abs(x).toString().split("").reverse().join('');
    const k = x<0 ? -1:1;
    y = y*k;
    if (y >= INT_MAX || y<= INT_MIN){
        return 0;
    }
    return y;
};

const x =-123
const y = reverse(x);
console.log(y);

File: C++/7_reverseInteger.cpp
#include <iostream>
#include <math.h> 

class Solution {
public:
    int reverse(int x) {
        int INT_MAX = pow(2,31) - 1;
        int INT_MIN = -pow(2,31);
        int rev = 0;
        while (x != 0) {
            int pop = x % 10;
            x /= 10;
            if (rev > INT_MAX/10 || (rev == INT_MAX / 10 && pop > 7)) return 0;
            if (rev < INT_MIN/10 || (rev == INT_MIN / 10 && pop < -8)) return 0;
            rev = rev * 10 + pop;
        }
        return rev;
    }
};

int main(void)
{
	Solution s;
	int res = s.reverse(-123);
    std::cout<<res<<std::endl;
}